packages/@aws-cdk/aws-sns-subscriptions/lib/sqs.ts

```diff
@@ -50,6 +50,15 @@ export class SqsSubscription implements sns.ITopicSubscription {
       protocol: sns.SubscriptionProtocol.SQS,
       rawMessageDelivery: this.props.rawMessageDelivery,
       filterPolicy: this.props.filterPolicy,
+      region: this.regionFromArn(topic),
     };
   }
+
+  private regionFromArn(topic: sns.ITopic): string | undefined {
+    if (!topic.topicArn || Token.isUnresolved(topic.topicArn)) {
```

Comment:
I could totally import an external topic and have it's topicArn be a token. I don't think that this is the right test here if the intention is to identify the local case. Why not just `topic instanceof sns.Topic`?

Reply:
Fair enough, I need to drop the comment here or update it.

Besides that, do you see any issue with the logic? To determine the region of the topic, I need the topic ARN to exist and be resolved. In the case that the ARN is unresolved, we can't determine the region to set.
Is there another way to determine region besides from the ARN?

Reply:
Unresolved ARNs are passable. Look at the logic of parseArn. It takes care of it.

Reply:
Goddamn! Missed that.